akka-projection-grpc/src/main/scala/akka/persistence/EventWriter.scala

```diff
@@ -0,0 +1,339 @@
+                  if (alreadyInJournal.isEmpty) {
+                    // error was not about duplicates
+                    state.waitingForReply.values.foreach {
+                      case (_, replyTo) =>
+                        replyTo ! StatusReply.error("Journal write failed")
+                    }
+                    context.log.warnN(
+                      "Failed writing event batch persistence id [{}], sequence nr [{}-{}]: {}",
+                      pid,
+                      sortedSeqs.head,
+                      sortedSeqs.last,
+                      originalErrorDesc)
+                    val newState = state.copy(waitingForReply = Map.empty)
+                    handleUpdatedStateForPid(pid, newState)
+                  } else {
+                    // ack all already written
+                    val stateAfterWritten = alreadyInJournal.foldLeft(state) { (state, seqNr) =>
+                      val (_, replyTo) = state.waitingForReply(seqNr)
+                      replyTo ! StatusReply.success(WriteAck(pid, seqNr))
+                      state.copy(waitingForReply = state.waitingForReply - seqNr)
+                    }
+                    if (needsWrite.isEmpty) {
+                      handleUpdatedStateForPid(pid, stateAfterWritten)
+                    } else {
+                      // retrigger write for those left if any
+                      val reprsToRewrite =
+                        stateAfterWritten.waitingForReply.values.map { case (repr, _) => repr }.toVector
+                      if (context.log.isDebugEnabled())
+                        context.log.debugN(
+                          "Partial batch was duplicates, re-triggering write of persistence id [{}], sequence nr [{}-{}]",
+                          pid,
+                          reprsToRewrite.head.sequenceNr,
+                          reprsToRewrite.last.sequenceNr)
+                      val write = AtomicWrite(reprsToRewrite) :: Nil
+                      journal ! JournalProtocol.WriteMessages(write, context.self.toClassic, actorInstanceId)
+                    }
+
+                  }
+
+              }
+              Behaviors.same
+
+            case response: JournalProtocol.Response => handleJournalResponse(response)
+
+            case unexpected =>
+              context.log.warn("Unexpected message sent to EventWriter [{}], ignored", unexpected.getClass)
+              Behaviors.same
+
+          }
+        })
+      .onFailure[Exception](SupervisorStrategy.restart)
+      .narrow[Command]
+
+  object MaxSeqNrFinder {
+    def apply(
+        replyTo: ActorRef[EventWriter.MaxSeqNrForPid],
+        journal: akka.actor.ActorRef,
+        persistenceId: String,
+        fromSeqNr: Long,
+        toSeqNr: Long,
+        originalErrorDesc: String): Behavior[Nothing] =
+      Behaviors
+        .setup[AnyRef] { context =>
+          journal ! JournalProtocol.ReplayMessages(
+            fromSeqNr,
+            toSeqNr,
```

Comment:
I think we can load the highest seqNr without replaying any events, and we only need the highest because then we know that all lower exist.

See AsyncWriteJournal:
```
            if (toSeqNr <= 0L || fromSequenceNr > toSeqNr)
              Future.successful(highSeqNr)
```

Reply:
Changed in 668e4dbf0904e7d8e1aad17e9aee210a9088bf9d

Reply:
not fully, as I intended, we still replay one event, right? I think this would do the trick:
```
fromSeqNr = 0,
toSeqNr = 0
```

Then it will not replay any messages but still reply with highestSeqNr.

With that, what do you think about skipping the actor completely and use `context.ask` instead?

Reply:
Ah, cap to 0, not max vaue, yes, if we know there are is actual replay it should be fine with ask 👍 neat.

Reply:
Seems to work nicely, changed in [9799279](https://github.com/akka/akka-projection/pull/932/commits/9799279daa99c9b3fae1d3ab2219f106d221fc0f)